src/ostree/ot-builtin-commit.c

```diff
@@ -219,6 +222,31 @@ commit_filter (OstreeRepo         *self,
   return OSTREE_REPO_COMMIT_FILTER_ALLOW;
 }
 
+typedef struct {
+  GRegex *regex;
+  const char *replacement;
+} TranslatePathnameData;
+
+/* Implement --tar-pathname-filter */
+static char *
+handle_translate_pathname (OstreeRepo *repo,
+                           GFileType ftype,
+                           const char *path,
+                           gpointer user_data)
+{
+  TranslatePathnameData *tpdata = user_data;
+  g_autoptr(GError) tmp_error = NULL;
+  char *ret =
+    g_regex_replace (tpdata->regex, path, -1, 0,
+                     tpdata->replacement, 0, &tmp_error);
+  if (!ret)
+    {
+      g_assert_no_error (tmp_error);
+      g_assert_not_reached ();
+    }
```

Comment:
If the replacement didn't make any changes, we should return `NULL`, no? At least so we test the fallback behaviour.